kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/filter/validation/JsonSchemaRecordValidationIT.java

```diff
@@ -268,7 +309,30 @@ void detectsClientProducingWithWrongSchemaId(boolean schemaIdInHeader, KafkaClus
             var bean = new PersonBean("john", "smith", 23);
             var invalid = producer.send(new ProducerRecord<>(topic1.name(), "my-key", bean));
             assertThatFutureFails(invalid, InvalidRecordException.class, "Unexpected schema id in record (1), expecting 2");
+        }
+    }
+
+    @ParameterizedTest
+    @ValueSource(booleans = { true, false })
+    void detectsClientProducingWithWrongKeySchemaId(boolean schemaIdInHeader, KafkaCluster cluster, Topic topic1) {
+        var config = proxy(cluster)
+                .addToFilters(new FilterDefinitionBuilder(RecordValidation.class.getName()).withConfig("rules",
+                        List.of(Map.of("topicNames", List.of(topic1.name()), "keyRule",
+                                Map.of("schemaValidationConfig", Map.of("apicurioRegistryUrl", APICURIO_REGISTRY_URL, "apicurioGlobalId", secondGlobalId)))))
+                        .build());
```

Comment:
feels like a stable setup type thing as well. If topic was annotated `@Name` we could inject the same one into setup as well as the test methods.

Reply:
Some of the tests use several topics so it might make things more confusing.

You could extract a method for the FilterDefinitionBuilder with topics as a var arg.